job/server/src/main/java/alluxio/master/job/JobMaster.java

```diff
@@ -319,6 +328,15 @@ public JobInfo getStatus(long jobId) throws JobDoesNotExistException {
     return new JobServiceSummary(jobInfos);
   }
 
+  /**
+   * @return health metrics for each of the job workers
+   */
+  public List<JobWorkerHealth> getAllWorkerHealth() {
+    ArrayList<JobWorkerHealth> result = Lists.newArrayList(mWorkerHealth.values());
+    Collections.sort(result, Comparator.comparingLong((a) -> a.getWorkerId()));
```

Comment:
This could be relatively expensive depending on the size of the cluster. Maybe we could use a `ConcurrentSkipListSet/Map` instead?

Reply:
Have you been able to do a performance comparison? Sorting every time someone does an RPC sounds more expensive to me than the extra bookkeeping required for using a ConcurrentSkipList, especially once a cluster grows to a few hundred or even thousands of workers.